pkg/setting/setting.go

```diff
@@ -787,26 +787,36 @@ func (cfg *Cfg) Load(args *CommandLineArgs) error {
 	}
 
 	// check old key  name
-	GrafanaComUrl = iniFile.Section("grafana_net").Key("url").MustString("")
+	GrafanaComUrl = valueAsString(iniFile.Section("grafana_net"), "url", "")
 	if GrafanaComUrl == "" {
-		GrafanaComUrl = iniFile.Section("grafana_com").Key("url").MustString("https://grafana.com")
+		GrafanaComUrl = valueAsString(iniFile.Section("grafana_com"), "url", "https://grafana.com")
 	}
 
 	imageUploadingSection := iniFile.Section("external_image_storage")
-	ImageUploadProvider = imageUploadingSection.Key("provider").MustString("")
+	ImageUploadProvider = valueAsString(imageUploadingSection, "provider", "")
 
 	enterprise := iniFile.Section("enterprise")
-	cfg.EnterpriseLicensePath = enterprise.Key("license_path").MustString(filepath.Join(cfg.DataPath, "license.jwt"))
+	cfg.EnterpriseLicensePath = valueAsString(enterprise, "license_path", filepath.Join(cfg.DataPath, "license.jwt"))
 
 	cacheServer := iniFile.Section("remote_cache")
 	cfg.RemoteCacheOptions = &RemoteCacheOptions{
-		Name:    cacheServer.Key("type").MustString("database"),
-		ConnStr: cacheServer.Key("connstr").MustString(""),
+		Name:    valueAsString(cacheServer, "type", "database"),
+		ConnStr: valueAsString(cacheServer, "connstr", ""),
 	}
 
 	return nil
 }
 
+func valueAsString(section *ini.Section, keyName string, defaultValue string) string {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Fatal(4, "Invalid value for key '%s' in configuration file", keyName)
```

Comment:
I dont want to use `log.Fatal` like this since it calls `os.Exit`. The function should return an error instead. `os.Exit` Should only be called from the `pkg/cmd/grafana-server` package

Reply:
Thanks for your feedback, i will replace `log.Fatal` with `error`.
I thought it was ok to use `log.Fatal` in `settings.go`, because it is already being used there.
Maybe we should open an issue to remove all usage of `os.Exit` and `log.Fatal` outside of `pkg/cmd/grafana-server` package?

Reply:
> I dont want to use `log.Fatal` like this since it calls `os.Exit`. The function should return an error instead. `os.Exit` Should only be called from the `pkg/cmd/grafana-server` package

I replaced `log.Fatal` with `error` and added a few tests.